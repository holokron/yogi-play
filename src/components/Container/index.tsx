export interface Props {
  children?: any;
}

export default function Container({ children }: Props): JSX.Element {
  return <div className="pt-4">{children}</div>;
}
