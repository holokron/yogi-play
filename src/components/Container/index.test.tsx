import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Container from '.'

describe('@components/Container', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <Container>Test</Container>
        ).toJSON()
        
        expect(tree).toMatchSnapshot()
    })

    it('should render without errors', () => {
        expect(
            shallow(
                <Container>Test</Container>
            ).matchesElement(
                <div className="pt-4 container">Test</div>
            )
        ).toBeTruthy()

    })

    it('should render without errors (fluid = true)', () => {
        expect(
            shallow(
                <Container fluid>Test</Container>
            ).matchesElement(
                <div className="pt-4 container-fluid">Test</div>
            )
        ).toBeTruthy()
    })
})