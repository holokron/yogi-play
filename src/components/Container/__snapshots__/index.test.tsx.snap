// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Container renders correctly 1`] = `
<div
  className="pt-4 container"
>
  Test
</div>
`;
